sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala

```diff
@@ -87,8 +101,8 @@ case class CreateHiveTableAsSelectCommand(
       }
     } else {
       try {
-        sparkSession.sessionState.executePlan(InsertIntoTable(
-          metastoreRelation, Map(), query, overwrite = true, ifNotExists = false)).toRdd
+        sparkSession.sessionState.executePlan(InsertIntoTable(metastoreRelation,
```

Comment:
oh, we should be fine here, the table is created with `reorderedOutputQuery.schema`, so there won't be any type difference